.github/workflows/test.yml

```diff
@@ -3,26 +3,44 @@ name: Build
 on:
   push:
     branches:
-      - master
+      - rewrite
   pull_request:
     branches:
-      - master
+      - rewrite
 
 jobs:
   build:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        node-version:
-          - '12'
-          - '14'
-          - '16'
+        node-version: ['14.x', '16.x']
+
     steps:
-     - uses: actions/checkout@v2
-     - uses: actions/setup-node@v2
-       with:
-         node-version: ${{ matrix.node-version }}
-     - name: Install dependencies
-       run: npm ci
-     - name: Run tests
-       run: npm t
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Cache modules
+        uses: actions/cache@v2
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+      - name: npm7
+        if: matrix.node-version != '16.x'
+        run: npm i -g npm@7
+      - name: Install
+        run: npm ci
+      - name: Lint
+        if: matrix.node-version == '16.x'
```

Comment:
Is this just for saving some CPU cycles (running it once will be enough) or is there some other reason to it?

Reply:
Yep. Build speed as well. An alternative could be running lint and prettier in a separate job, but then it needs a to run npm install again.

Reply:
I wonder if this could turn into a red herring for people that don't know about this and have a build that fails on Node 16, but passes on Node 14, and who then go spelunking in changelogs and everything instead of having a look at the build output.

Probably ok to do it like this still.